__tests__/index.test.js

```diff
@@ -15,41 +16,77 @@ afterAll(() => {
 describe('General bad requests', () => {
   it('undefined endpoint', () => {
     return request(app)
-    .get('/api/pigs')
-    .expect(404)
+      .get('/api/pigs')
+      .expect(404)
   })
 })
 
 describe('GET /api/topics', () => {
   it('should return an array of objects, each having a slug and description property', () => {
     return request(app)
-    .get('/api/topics')
-    .expect(200)
-    .then((res) => {
-      const topics = res.body.topics
-      expect(topics.length).toBe(3)
-      topics.forEach((topic) => {
-        expect(topic).toMatchObject({
-          slug: expect.any(String),
-          description: expect.any(String)
+      .get('/api/topics')
+      .expect(200)
+      .then((res) => {
+        const topics = res.body.topics
+        expect(topics.length).toBe(3)
+        topics.forEach((topic) => {
+          expect(topic).toMatchObject({
+            slug: expect.any(String),
+            description: expect.any(String)
+          })
         })
       })
-    })
   })
 })
 
 describe('GET /api', () => {
   it('should return all available endpoints', () => {
+    const actualFile = require('../endpoints.json')
     return request(app)
-    .get('/api')
-    .then((res) => {
-      expect(() => JSON.parse(res.text)).not.toThrow(new Error)
-      const endpoints = JSON.parse(res.text).endpoints
-      Object.keys(endpoints).forEach((key) => {
-        expect(endpoints[key]).toMatchObject({
-          description: expect.any(String)
+      .get('/api')
+      .then((res) => {
+        expect(() => JSON.parse(res.text)).not.toThrow(new Error)
+        const endpoints = JSON.parse(res.text).endpoints
+        // exception as this is actually what is requested.
+        delete endpoints['GET /api']
+        Object.keys(endpoints).forEach((key) => {
+          expect(endpoints[key]).toMatchObject({
+            description: expect.any(String),
+            queries: expect.any(Array),
+            exampleResponse: expect.any(Object)
+          })
         })
+        expect(actualFile).toEqual(JSON.parse(res.text).endpoints)
       })
-    })
+  })
+})
+
+describe('GET /api/articles/:article_id', () => {
+  it('should return an article by its ID', () => {
+    const urlPattern = /(?:https?):\/\/(\w+:?\w*)?(\S+)(:\d+)?(\/|\/([\w#!:.?+=&%!\-\/]))?/
+    return request(app)
+      .get('/api/articles/2')
+      .then((res) => {
+        const article = res.body.article
+        expect(article.length).toBe(1)
```

Comment:
We should not be expecting article to be an array. This should be the article object itself.